text/0000-stack-buildpacks.md

```diff
@@ -0,0 +1,596 @@
+# Meta
+[meta]: #meta
+- Name: Stack Buildpacks
+- Start Date: 2020-08-27
+- Author(s): [@jkutner](@jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is proposal for a new type of buildpack that runs against a stack in order to extend it in ways that are only possible by running privileged commands.
+
+# Motivation
+[motivation]: #motivation
+
+Normally, buildpacks do not run as root. This is an intentional design decision that ensures operations like `rebase` will work on day-2.
+
+However, many applications and buildpacks require modifications to the stack they run on, such as adding system packages or custom certificates. For this reason, we need a mechanism that buildpack authors and buildpack users and leverage to extend their stacks.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- *root buildpack* - a new type of buildpack that runs as the root user
+- *stack buildpack* - a type of root buildpack that runs against the stack image(s) instead of an app. Stack buildpacks must not make changes to the build and run images that either violate stack [compatibility guarantees](https://github.com/buildpacks/spec/blob/main/platform.md#compatibility-guarantees) or violate the contract defined by that stack's author.
+- *userspace buildpack* - the traditional definition of a buildpack (i.e. does not run as root, and runs against an app)
+
+A new type of buildpack, called a Stack buildpack, may run against a stack (both build and run images) in order to extend it in ways that are only possible by running privileged commands. Unlike userspace buildpacks, stack buildpack can modify any path on the filesystem. Userspace buildpack can only create/modify disjoint layers (either by adding a dir to `<layers>` or modifying an app slice), which makes possible features like individual layer reuse that is independent or ordering.
```

Comment:
The first part of the sentence looks like a redefinition of line 27. Maybe you can combine both definitions together?

Reply:
i tried reworking this, but it was hard to avoid _some_ repetition between the definition and the problem statement. I'm gonna keep as-is, but feel free make an edit suggestion.